agent/xds/routes.go

```diff
@@ -61,6 +61,11 @@ func (s *ResourceGenerator) routesForConnectProxy(cfgSnap *proxycfg.ConfigSnapsh
 			continue
 		}
 
+		if !structs.IsProtocolHTTPLike(chain.Protocol) {
```

Comment:
This same check happens below when creating routes for Destination Upstreams: https://github.com/hashicorp/consul/pull/19496/files#diff-d98ce89d93316f2ab4fd8ea0561b9121e88e535567b9fd4d0992671d145b4b85R93

The two tests that were moved to resource_test.go in this PR both hit this condition where a TCP discovery chain creates a route resource that is not referenced by a listener filter chain.